packages/appkit/src/client.ts

```diff
@@ -1406,10 +1410,25 @@ export class AppKit {
     const adapter = this.getAdapter(ChainController.state.activeChain as ChainNamespace)
 
     this.chainNamespaces.forEach(async chainNamespace => {
-      const caipAddress = this.universalProvider?.session?.namespaces?.[chainNamespace]
-        ?.accounts[0] as CaipAddress
+      const namespaceAccounts =
+        this.universalProvider?.session?.namespaces?.[chainNamespace]?.accounts || []
+
+      // We try and find the address for this network in the session object.
+
+      const activeCaipNetworkId = ChainController.state.activeCaipNetwork?.id
+
+      const caipAddress =
+        namespaceAccounts.find(account => {
+          const [, chainId] = account.split(':')
+
+          return chainId === activeCaipNetworkId?.toString()
+        }) || namespaceAccounts[0]
 
       if (caipAddress) {
+        if (caipAddress.split(':').length !== 3) {
```

Comment:
Seems like we could have a utility in common (or another pck) for this instead of doing here, wdyt? We have `CaipNetworkUtil` in `appkit-utils`